### DISCUSSION TITLE
How to intercept the API when mulltiple instance of API exist in the same name

### DISCUSSION THREAD

**Discussion by jeevankumar22** [getgauge/taiko#2250] [General]
End point : https://example.com/questions/config
API 	  : config

Scenario : 

When i navigate to my application two API call exist in the same ( config ) , i.e two instance of API with same name

the difference between this two API call is payload 

Below is the sample payload of these two instance of API 

API call 1 : 
Payload : 
types: ["modal", "tab", "mobile number", "name", "cost"]

API call 2 : 
Payload : ["name", "address"]

Challenge : 
When i try to intercept from the tool, its always intercepting the first API call / instance
How do we instruct the taiko to mock the second instance of API / how to instruct taiko to check the header for all the instance of API and intercept the API based on the certain condition

I followed the available methods from the documentation , but couldn't solve this problem , taiko looking API with name , once matching instance is found its trying to intercept, but in my case i want to iterate all the instance of API call and based on the condition it should intercept

**Comment by zabil** [comment]
Have you tried using case 5 at https://docs.taiko.dev/api/intercept/ ? i.e

```
# Case 5: Mock response based on a request
await intercept(url, (request) => { request.respond({mockResponseObject}) })
```

You can add a condition by inspecting the request.

**Reply by jeevankumar22** [reply]
yes i tried this method too..
here it's checking request , but while checking the request it's considering the API call from the first instance..
It's iterating to all the existing instance of API with same name

**Reply by zabil** [reply]
Ok, right [now](https://github.com/getgauge/taiko/blob/4eb71ac406932a8b1ec4859da6cb18067b652a48/lib/handlers/fetchHandler.js) Taiko's `intercept` API only intercepts the first occurrence of a matching url and ignores other occurences. On run, Taiko displays a warning. So there's no way to match the second occurrence of the same url.
